scripts/install/install-helper-functions.sh

```diff
@@ -78,15 +81,65 @@ function proc_net_route_addr() {
   printf "%d.%d.%d.%d\n" "${a}" "${b}" "${c}" "${d}"
 }
 
-# Print the first IP on the route to 8.8.8.8, i.e. to Google's public DNS server.
-# Will be one of this hosts IP addresses or nothing if 'ip' is not installed.
-function route_dns_addr() {
-  if [ -x /sbin/ip ] ; then
-    # What is the route to 8.8.8.8, i.e. to Google's public DNS server?
-    /sbin/ip route get 8.8.8.8 | awk -F"src " 'NR==1{split($2,a," ");print a[1]}'
+# Print the address of the first gateway on the route to 8.8.8.8, i.e. to
+# Google's public DNS server. Inside a docker container, this will be the
+# address of the host. Requires the iproute2 package be installed, which it
+# is not by default in an ubuntu-18.04 docker image.
+function sbin_ip_gateway() {
+  local cmd
+  if [ -x "$(safe_which ip)" ] ; then
+    cmd='ip'
+  elif [ -x /sbin/ip ] ; then
+    cmd='/sbin/ip'
+  else
+    return 0
+  fi
+  "${cmd}" route get 8.8.8.8 | awk -F"via " 'NR==1{split($2,a," ");print a[1]}'
+}
+
+# Try all methods, output all gateway addresses (there could be duplicates).
+function get_gateway_ips() {
+  proc_net_route_gateway
+  sbin_ip_gateway
+}
+
+#-------------------------------------------------------------------------------
+# Helpers for finding the IP address of this device.
+
+# Get the IP addresses of the host, of which there may be multiple or none.
+function hostname_addresses() {
+  if [ -x "$(safe_which hostname)" ] ; then
+    hostname --all-ip-addresses
+  fi
+}
+
+# Print the first IP on the route to 8.8.8.8, i.e. to Google's public DNS
+# server. Inside a docker container, this will be the address of the container,
+# and on a host, this will be the address of one of the interfaces or nothing
+# if 'ip' is not installed or there is no interfaces enabled.
+# Requires the iproute2 package be installed, which it is not by default in an
+# ubuntu-18.04 docker image.
+function sbin_ip_host() {
+  local cmd
+  if [ -x "$(safe_which ip)" ] ; then
+    cmd='ip'
+  elif [ -x /sbin/ip ] ; then
+    cmd='/sbin/ip'
+  else
+    return 0
   fi
+  "${cmd}" route get 8.8.8.8 | awk -F"src " 'NR==1{split($2,a," ");print a[1]}'
```

Comment:
Seems like the only difference from `sbig_ip_gateway` is the work `src` here instead of `via`. Does it make sense to combine these two?

Reply:
Agreed, that is only difference. Extracted out.